geode-core/src/main/java/org/apache/geode/distributed/internal/ReplyMessage.java

```diff
@@ -309,11 +316,17 @@ public void fromData(DataInput in,
     if (testFlag(status, PROCESSOR_ID_FLAG)) {
       this.processorId = in.readInt();
     }
-    if (testFlag(status, EXCEPTION_FLAG)) {
-      this.returnValue = DataSerializer.readObject(in);
-      this.returnValueIsException = true;
-    } else if (testFlag(status, OBJECT_FLAG)) {
-      this.returnValue = DataSerializer.readObject(in);
+    try {
+      if (testFlag(status, EXCEPTION_FLAG)) {
+        this.returnValue = context.getDeserializer().readObject(in);
+        this.returnValueIsException = true;
+      } else if (testFlag(status, OBJECT_FLAG)) {
+        this.returnValue = context.getDeserializer().readObject(in);
+        this.returnValueIsException = (returnValue instanceof ReplyException);
+      }
+    } catch (IOException e) {
```

Comment:
In `toData()` the comment says we will get a `WriteAbortedException`.

Would it be better to catch exactly that? Is there a reason why catching the more general class is better? I honestly don't know the answer, but was hoping I might learn something by asking.

Reply:
I could narrow it down, but it was very surprising to me that a "read" operation would throw a WriteAbortedException and I wanted to cover all the bases.

Reply:
I like that answer @bschuchardt !